File: app/src/main/java/dev/yuganshtyagi/smileyratingexample/MainActivity.java
package dev.yuganshtyagi.smileyratingexample;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.widget.RatingBar;

import dev.yuganshtyagi.smileyrating.SmileyRatingView;

import com.yugansh.tyagi.smileyratingexample.R;

public class MainActivity extends AppCompatActivity {

    RatingBar ratingBar;
    SmileyRatingView smileyRating;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        smileyRating = findViewById(R.id.smiley_view);

        ratingBar = findViewById(R.id.rating_bar);
        ratingBar.setOnRatingBarChangeListener(new RatingBar.OnRatingBarChangeListener() {
            @Override
            public void onRatingChanged(RatingBar ratingBar, float rating, boolean fromUser) {
                Log.d("Rating", String.valueOf(rating));
                smileyRating.setSmiley(rating);
            }
        });
    }
}

File: app/src/main/java/dev/yuganshtyagi/smileyratingexample/MainActivity.kt
package dev.yuganshtyagi.smileyratingexample

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.yugansh.tyagi.smileyratingexample.R.layout
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layout.activity_main)

        rating_bar.setOnRatingBarChangeListener { _, rating, _ ->
            smiley_view.setSmiley(rating = rating)
        }
    }
}